netbeans/CheckerFrameworkPlugin/src/org/checkerframework/netbeans/CheckerFrameworkPanel.java

```diff
@@ -0,0 +1,159 @@
+
+        try {
+            editableProperty = loadProperties(projectProperties);
+        } catch (IOException e) {
+            System.out.println("Failed to load netbeans project.properties file.");
+        }
+
+        try {
+            checkerStrings =
+                    loadProperties(
+                                    FileUtil.toFileObject(
+                                            InstalledFileLocator.getDefault()
+                                                    .locate(
+                                                            "checkerstrings.properties",
+                                                            "org.checkerframework.netbeans",
+                                                            false)))
+                            .entrySet();
+        } catch (IOException e) {
+            System.out.println("Failed to load checker strings properties file.");
+        }
+
+        checkerList = new JCheckBox[checkerStrings.size()];
+        int i = 0;
+        selection =
+                new StringBuilder(editableProperty.get("annotation.processing.processors.list"));
+        for (Entry<String, String> e : checkerStrings) {
+            checkerList[i] = new JCheckBox(e.getValue());
+            if (selection.toString().contains(e.getKey())) {
+                checkerList[i].setSelected(true);
+            }
+            this.add(checkerList[i]);
+            checkerList[i].addItemListener(new CheckBoxListener(selection));
+            i++;
+        }
+    }
+
+    /**
+     * Helper method to load a properties file
+     *
+     * @param propsFO File object to be loaded
+     * @return EditableProperties class loaded with the properties file
+     * @throws IOException
+     */
+    private static EditableProperties loadProperties(FileObject propsFO) throws IOException {
+        InputStream propsIS = propsFO.getInputStream();
+        EditableProperties props = new EditableProperties(true);
+        try {
+            props.load(propsIS);
+        } finally {
+            propsIS.close();
+        }
+        return props;
+    }
+
+    /**
+     * This method is called by the action listener that runs when the Project Properties window is
+     * closed and will store the selected checkers into the project.properties file.
+     *
+     * @throws IOException
+     */
+    public void store() throws IOException {
+        try {
+            ProjectManager.mutex()
+                    .writeAccess(
+                            new WriteCheckerFrameworkPropertiesAction(
+                                    projectProperties, checkerPath, checkerQualPath, selection));
+        } catch (MutexException mux) {
+            throw (IOException) mux.getException();
+        }
+    }
+
+    /** ItemListener that updates the checkers to run based on the check boxes that get selected. */
+    private static class CheckBoxListener implements ItemListener {
+        private StringBuilder s;
```

Comment:
The frequent modifications of the StringBuilder when the checkbox is selected and deselected seems very awkward.
Instead, you could simply maintain a `Set<String>` that contains the currently selected checkers.
Then only once you need the final String, you concatenate the elements of the set.